pkg/sql/job_statistics.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sql
+
+import (
+	"context"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/retry"
+)
+
+type metricsPoller struct {
```

Comment:
why not define this job internally, in the jobs package? 
Or is there some circular dependency issues you're trying to work around?

Reply:
Yes. I need `sql.JobExecContext` in `Resume`, so `Resume` must be outside the package. Then, I figured I should keep the struct in the same file.